### PR TITLE
Add compressed writes to cas.go.

### PR DESCRIPTION
**Pull request by rubensf**
This follows the current tentative API. This PR potentially replaces #232

### CODE REVIEW

File: go/pkg/client/bytestream.go [hunk 1]
```diff
@@ -42,6 +42,10 @@ func (c *Client) WriteChunked(ctx context.Context, name string, ch *chunker.Chun
 				return err
 			}
 			if chunk.Offset == 0 {
+				// Notice that the digest in the chunker might be misleading.
+				// Specifically, for compressed blob uploads, the resource
+				// name should include the uncompressed digest - while chunker
+				// should be including the compressed digest.
```
Comment (ola-rozenfeld) [inline go/pkg/client/bytestream.go:48]: I thought that you removed the Digest() function from Chunker altogether, instead relying on UE.digest

File: go/pkg/client/tree.go [hunk 2]
```diff
@@ -65,7 +65,7 @@ func shouldIgnore(inp string, t command.InputType, excl []*command.InputExclusio
 
 // loadFiles reads all files specified by the given InputSpec (descending into subdirectories
 // recursively), and loads their contents into the provided map.
-func loadFiles(execRoot string, excl []*command.InputExclusion, path string, fs map[string]*fileSysNode, cache filemetadata.Cache) error {
+func (c *Client) loadFiles(execRoot string, excl []*command.InputExclusion, path string, fs map[string]*fileSysNode, cache filemetadata.Cache) error {
```
Comment (ola-rozenfeld) [inline go/pkg/client/tree.go:68]: Where are you using the client here?

Comment (rubensf) [reply go/pkg/client/tree.go]: I shouldn't - this was a bad rebasing.

File: go/pkg/client/bytestream.go (hunk 1 above)
Comment (rubensf) [reply go/pkg/client/bytestream.go:48]: Yes - this is just an artifact of rebasing :)

**Comment by rubensf** [comment]
@googlebot I fixed it.

File: go/pkg/client/cas.go [hunk 3]
```diff
@@ -1377,3 +1387,14 @@ func (c *Client) DownloadFiles(ctx context.Context, execRoot string, outputs map
 	}
 	return nil
 }
+
+func (c *Client) shouldCompress(sizeBytes int64) bool {
```
Comment (ola-rozenfeld) [inline go/pkg/client/cas.go:1391]: nit: If the function is not reused, maybe inline?

File: go/pkg/client/cas_test.go [hunk 4]
```diff
@@ -712,74 +714,71 @@ func TestUpload(t *testing.T) {
...
-						present := make(map[digest.Digest]bool)
-						for _, blob := range tc.present {
-							fake.Put(blob)
-							present[digest.NewFromBlob(blob)] = true
-						}
-						var input []*uploadinfo.Entry
-						for _, blob := range tc.input {
-							input = append(input, uploadinfo.EntryFromBlob(blob))
-						}
+			for _, cmp := range []client.CompressedBytestreamThreshold{-1, 0} {
```
Comment (ola-rozenfeld) [inline go/pkg/client/cas_test.go:717]: Note to self: add a []client.Opt field to tests and expand the tests to contain the cartesian product of these options in a loop before running them. The indentation is getting out of hand...
(I'll clean this up separately)

**Review by ola-rozenfeld — APPROVED** [review]

File: go/pkg/client/cas.go (hunk 3 above)
Comment (rubensf) [reply go/pkg/client/cas.go:1391]: I'll be reusing it on read compression :)

File: go/pkg/client/cas_test.go (hunk 4 above)
Comment (rubensf) [reply go/pkg/client/cas_test.go:717]: sgtm - if the indentation is the only problem you can also move the inner block into a separate function. Too late for it right now, but it is useful whereas in it preserves the original blame!

**Pull request merged**